src/config.rs

```diff
@@ -342,6 +342,10 @@ impl Cfg {
         )
     }
 
+    pub fn get_default_opt(&self) -> Result<Option<String>> {
+        self.settings_file.with(|s| Ok(s.default_toolchain.clone()))
+    }
+
```

Comment:
Is `get_default` used a lot? I'd prefer to replace it rather than add another method

Reply:
Is not used in many places, I'll shuffle that around.  Good catch.